test/test_nn.py

```diff
@@ -20602,18 +20602,36 @@ def my_pre_load_hook_with_module(
                 nonlocal hook_called
                 hook_called += 1
 
-        m = MyModule()
-        state_dict = m.state_dict()
+        # Test that hooks registered on a submodule are also called
+        # appropriately, i.e. with the submodule as module argument in
+        # my_pre_load_hook_with_module.
+        class MyModuleContainer(nn.Module):
+            def __init__(self, mod):
+                super().__init__()
+                self.mod = mod
+
+        submodule = MyModule()
+        for i, m in enumerate([MyModuleContainer(submodule), submodule]):
```

Comment:
Curious what the intended purpose of the `i = 1` iteration is. It seems like it is testing that we can register multiple pre-hooks (without modules) and can register multiple pre-hooks with modules. Otherwise, it reduces to not testing a submodule but just a top-level module.

It was not immediately clear to me that `hook_called_offset` is there exactly because we are double-registering hooks on `submodule`. If we want to test multi-registering of hooks, then maybe we should separate it or explain this is what the loop is doing?